go-runtime/ftl/ftltest/ftltest.go

```diff
@@ -322,3 +326,38 @@ func WithCallsAllowedWithinModule() Option {
 		return nil
 	}
 }
+
+// WhenMap injects a fake implementation of a Mapping function
+//
+// To be used when setting up a context for a test:
+//
+//	ctx := ftltest.Context(
+//		ftltest.WhenMap(Example.MapHandle, func(ctx context.Context, t T) (U, error) {
+//	    	// ...
+//		}),
+//		// ... other options
+//	)
+func WhenMap[T, U any](mapper *ftl.MapHandle[T, U], fake func(context.Context, T) (U, error)) Option {
+	return func(ctx context.Context, state *OptionsState) error {
+		ref := reflection.TypeRefFromValue(mapper)
```

Comment:
You don't really need to use a `Ref` as the key, `Map` is not really an FTL concept, it's just a helper function in the Go runtime.

Reply:
replaced with uintptr